resources/helm/v2.1/istio-control/istio-discovery/templates/mutatingwebhook.yaml

```diff
@@ -9,26 +33,38 @@ metadata:
     app: sidecar-injector
     release: {{ .Release.Name }}
 webhooks:
-  - name: sidecar-injector.istio.io
-    clientConfig:
-      {{- if .Values.istiodRemote.injectionURL }}
-      url: {{ .Values.istiodRemote.injectionURL }}
-      {{- else }}
-      service:
-        name: istiod-{{ .Values.revision | default "default" }}
-        namespace: {{ .Release.Namespace }}
-        path: "/inject"
-      {{- end }}
-      caBundle: ""
-    sideEffects: None
-    rules:
-      - operations: [ "CREATE" ]
-        apiGroups: [""]
-        apiVersions: ["v1"]
-        resources: ["pods"]
-    failurePolicy: Fail
-    admissionReviewVersions: ["v1beta1", "v1"]
-    namespaceSelector:
+{{- if .Values.sidecarInjectorWebhook.useLegacySelectors}}
+{{- /* Setup the "legacy" selectors. These are for backwards compatibility, will be removed in the future. */}}
+{{- include "core" (mergeOverwrite (deepCopy .) (dict "Prefix" "")) }}
+  namespaceSelector:
+      matchExpressions:
+      - key: maistra.io/member-of
+        operator: In
+        values:
+        - {{ .Release.Namespace }}
+      - key: maistra.io/ignore-namespace
+        operator: DoesNotExist
+      - key: istio-injection
+        operator: NotIn
+        values:
+        - disabled
+      - key: istio-env
+        operator: DoesNotExist
+{{- if .Values.sidecarInjectorWebhook.objectSelector.enabled }}
+    objectSelector:
+      matchExpressions:
+{{- if eq .Values.global.proxy.autoInject "enabled" }}
+      - key: "sidecar.istio.io/inject"
+        operator: NotIn
+        values:
+        - "false"
+{{- else }}
+      - key: "sidecar.istio.io/inject"
+        operator: In
+        values:
+        - "true"
+{{- end }}
+{{- end }}
```

Comment:
These changes don't look correct.

Reply:
Sorry can you please elaborate?

Reply:
This is fixed.

Reply:
are you sure? it stil looks strange -- if `.Values.sidecarInjectorWebhook.useLegacySelectors` is enabled ,it will produce an invalid resource

Reply:
or maybe I'm misunderstanding how the mergeOverwrite above works

Reply:
Added https://github.com/maistra/istio-operator/pull/756 to fix the issue.